FXyz-Samples/src/main/java/org/fxyz3d/controls/TimelineControl.java

```diff
@@ -0,0 +1,181 @@
+ * derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL F(X)yz BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */ 
+
+package org.fxyz3d.controls;
+
+import javafx.animation.Timeline;
+import javafx.beans.binding.Bindings;
+import javafx.beans.property.*;
+import javafx.beans.value.ChangeListener;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ToggleButton;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Pane;
+import javafx.scene.layout.StackPane;
+import javafx.scene.text.Text;
+import javafx.util.Duration;
+
+/**
+ *
+ * @author Jose Pereda
+ */
+public class TimelineControl extends ControlBase<Property<Timeline>> {
+
+    @FXML private Label title;
+
+    @FXML private StackPane timelineDisplay;
+    @FXML private Pane background;
+    @FXML private HBox progressBar;
+    @FXML private Pane bar;
+    @FXML private Text startTxt;
+    @FXML private Text currentTxt;
+    @FXML private Text endTxt;
+
+    @FXML private Button startBtn;
+    @FXML private Button rwBtn;
+    @FXML private ToggleButton playBtn;
+    @FXML private Button ffBtn;
+    @FXML private Button endBtn;
+    @FXML private ToggleButton loopBtn;
+
+    private final ChangeListener<Number> rateListener = (obs, ov, nv) -> {
+        if (nv.intValue() == 0 && playBtn.isSelected()) {
+            playBtn.setSelected(false);
+        }
+    };
+
+    private final DoubleProperty currentTimeAsPercentage = new SimpleDoubleProperty(0);
+
+    public final Timeline getTimeline() { return timeline.get(); }
+
+    private final ObjectProperty<Timeline> timeline = new SimpleObjectProperty<>() {
+        private Timeline old;
+        @Override protected void invalidated() {
+            Timeline t = get();
+            if (old != null) {
+                currentTimeAsPercentage.unbind();
+                currentTxt.textProperty().unbind();
+                endTxt.textProperty().unbind();
+                bar.prefWidthProperty().unbind();
+                old.currentRateProperty().removeListener(rateListener);
+            }
+            if (t == null) {
+                timelineDisplay.setVisible(false);
+                startBtn.setDisable(true);
```

Comment:
Instead of `setDisable(true / false)` for buttons, perhaps we can do smth like: (I don't remember the exact method call) `startBtn.disableProperty().bind(timeline.isNull())`

Reply:
There is no such binding. I'll enable/disable the HBox parent of all those buttons instead.

Reply:
I have used [isNull()](https://docs.oracle.com/javase/8/javafx/api/javafx/beans/binding/ObjectExpression.html#isNull--)